velox/docs/functions/presto/regexp.rst

```diff
@@ -18,6 +18,11 @@ See https://github.com/google/re2/wiki/Syntax for more information.
     Note: The wildcard '%' represents 0, 1 or multiple characters and the
     wildcard '_' represents exactly one character.
 
+    Note: Each function instance allow for a maximum of 20 regular expressions to
+    be compiled throughout the lifetime of the query. Not all Patterns requires
+    compilation of regular expressions; for example a pattern 'aa' does not.
```

Comment:
It would be nice to document all patterns that do not use regex.

Reply:
yeh maybe we can ping the original author, even i do not understand all of them . I will create a github issue for that and add you and the author of the optimizations